rasa/core/policies/rule_policy.py

```diff
@@ -621,11 +698,43 @@ def train(
             trackers_as_states, trackers_as_actions
         )
 
+    def train(
+        self,
+        training_trackers: List[TrackerWithCachedStates],
+        domain: Domain,
+        interpreter: NaturalLanguageInterpreter,
+        **kwargs: Any,
+    ) -> None:
+        """Trains the policy on given training trackers.
+
+        Args:
+            training_trackers: The list of the trackers.
+            domain: The domain.
+            interpreter: Interpreter which can be used by the polices for featurization.
+        """
+        # only consider original trackers (no augmented ones)
+        training_trackers = [
+            t
+            for t in training_trackers
+            if not hasattr(t, "is_augmented") or not t.is_augmented
```

Comment:
why do we need `hasattr`?

Reply:
this is an old code. it happens because right now 2 types of trackers are not interchangeable because of this param. It needs to be fixed, but it is out of scope of this PR

Reply:
Is there an issue which describes the problem? And which two types of tracker?

Reply:
I don't think there is an issue.

> And which two types of tracker?

`TrackerWithCachedStates` and `DialogueStateTracker`